Form/DataTransformer/EntitiesToPropertyTransformer.php

```diff
@@ -107,19 +107,18 @@ public function reverseTransform($values)
             }
         }
 
-        try {
-          // get multiple entities with one query
-          $entities = $this->em->createQueryBuilder()
-              ->select('entity')
-              ->from($this->className, 'entity')
-              ->where('entity.'.$this->primaryKey.' IN (:ids)')
-              ->setParameter('ids', $values)
-              ->getQuery()
-              ->getResult();
-        }
-        catch (\Exception $ex) {
+        // get multiple entities with one query
+        $entities = $this->em->createQueryBuilder()
+            ->select('entity')
+            ->from($this->className, 'entity')
+            ->where('entity.'.$this->primaryKey.' IN (:ids)')
+            ->setParameter('ids', $values)
+            ->getQuery()
+            ->getResult();
+
           // this will happen if the form submits invalid data
-          throw new TransformationFailedException('One or more id values are invalid');
+        if (count($entities) != count($values)) {
+            throw new TransformationFailedException('One or more id values are invalid');
         }
 
```

Comment:
I think we still need to catch the exception in case the query fails.  We can also do the count comparison but I think we need both.

Reply:
> I think we still need to catch the exception in case the query fails. We can also do the count comparison but I think we need both.

If the query fails, it must be some config errors, so I think throw the exception to developer is ok.

Reply:
I think this issue https://github.com/tetranz/select2entity-bundle/issues/25https://github.com/tetranz/select2entity-bundle/issues/25 is the reason why we catch exceptions here and effectively convert them to TransformationFailedException.

There are situation related to tags where it is possible for the user to enter something that will cause the query to fail. We want Symfony to catch it as a TransformationFailedException so it will display `This value is not valid`.

Reply:
> I think this issue #25 is the reason why we catch exceptions here and effectively convert them to TransformationFailedException.
> 
> There are situation related to tags where it is possible for the user to enter something that will cause the query to fail. We want Symfony to catch it as a TransformationFailedException so it will display `This value is not valid` rather an uncaught exception.

No, I don't think so. If multiple is false,  getSingleResult will throw UnexpectedResultException ,  and I have changed the \Exception to it. If multiple is true,  getResult will return empty array, instead of throw Exception, so catch \Exception is useless, I add another judgment for that.